From: Dmitry Koroban <dkoroban@oktetlabs.ru>
To: hostap@lists.infradead.org
Subject: Hostapd does not update WPS credentials on SIGHUP
Date: Thu, 20 Aug 2020 13:46:57 +0300
Message-ID: <fdadce6eddc483bb5f4cd1a6a343e046@oktetlabs.ru>

Faced this issue on v2.6, but 2.10 seems to have it too.
Steps to reproduce:
1. Start hostapd with WPS enabled
2. Change passphrase in hostapd.conf
3. Send SIGHUP to hostapd
4. Try to connect client using wps_pbc
5. Client receives old passphrase and cannot connect

Signed-off-by: Dmitry Koroban <dkoroban at oktetlabs.ru>

Origin: https://patchwork.ozlabs.org/project/hostap/patch/fdadce6eddc483bb5f4cd1a6a343e046@oktetlabs.ru/
---

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index b37f49f9a..9c78785c6 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -155,6 +155,9 @@ static void hostapd_reload_bss(struct hostapd_data *hapd)
 		wpa_printf(MSG_ERROR, "Could not set SSID for kernel driver");
 		/* try to continue */
 	}
+	/* Update WPS credentials */
+	hostapd_deinit_wps(hapd);
+	hostapd_init_wps(hapd, hapd->conf);
 	wpa_printf(MSG_DEBUG, "Reconfigured interface %s", hapd->conf->iface);
 }
 
